xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/page/PageNotificationEventDescriptorContainer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.page;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Send a {@link PageNotificationEvent} when a custom event is triggered.
+ *
+ * @version $Id$
+ * @since 9.4RC1
+ */
+@Role
+@Unstable
+public interface PageNotificationEventDescriptorContainer
+{
+    /**
+     * Update the object descriptorList.
+     * @param descriptorList the new list of descriptors to apply
+     */
+    void updateDescriptorList(List<PageNotificationEventDescriptor> descriptorList);
```

Comment:
This method is meant to bind every given descriptors to a `RecordableEventDescriptorContainer` (as the descriptors are not components, we can’t use `@Inject`) and also "un-bind" them from the `RecordableEventDescriptorContainer` when a new list is given ("un-binding" them make them recoverable by the garbage collector). 
For implementation, see : https://github.com/xwiki/xwiki-platform/pull/570/files/080eefd2bafa1a041273631d2aac22794e4ea8ad#diff-63bd246df6cd989520f8379694756f7eR51